src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -248,8 +245,7 @@ public void Abort()
                         ResponseBody = _responseBody;
                         DuplexStream = new FrameDuplexStream(RequestBody, ResponseBody);
 
-                        _requestAbortCts = CancellationTokenSource.CreateLinkedTokenSource(_disconnectCts.Token);
-                        RequestAborted = _requestAbortCts.Token;
+                        RequestAborted = _disconnectOrAbortedCts.Token;
```

Comment:
It should activate for disconnects and Abort, and Abort also disconnects. It tells the application to abandon any processing and unwind the middleware chain.